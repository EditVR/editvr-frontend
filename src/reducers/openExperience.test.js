/**
 * @file experiences.test.js
 * Contains tests for experiences.js.
 */

import reducer from './openExperience';

import {
  OPEN_EXPERIENCE_FETCH_FOR_USER,
  OPEN_EXPERIENCE_SCENE_CREATE,
  OPEN_EXPERIENCE_SCENE_EDIT,
  OPEN_EXPERIENCE_COMPONENT_FIELD_PRESAVE,
  OPEN_EXPERIENCE_COMPONENT_EDIT
} from '../constants';

describe('reducers->openExperience', () => {
  it('Should return the initial state by default', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_FETCH_FOR_USER}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_FETCH_FOR_USER}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_FETCH_FOR_USER}_FAIL`, () => {
    const error = 'Error: failed to fetch experiences.';
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_FETCH_FOR_USER}_FAIL`,
        loading: false,
        payload: { error }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_FETCH_FOR_USER}_SUCCESS`, () => {
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_FETCH_FOR_USER}_SUCCESS`,
        loading: false,
        error: null,
        payload: [
          {
            id: 1,
            title: 'example experience'
          }
        ]
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_SCENE_CREATE}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_SCENE_CREATE}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_SCENE_CREATE}_FAIL`, () => {
    const error = 'Error: failed to create scene.';
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_SCENE_CREATE}_FAIL`,
        loading: false,
        payload: { error }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_SCENE_CREATE}_SUCCESS`, () => {
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_SCENE_CREATE}_SUCCESS`,
        loading: false,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_SCENE_EDIT}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_SCENE_EDIT}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_SCENE_EDIT}_FAIL`, () => {
    const error = 'Error: failed to update scene.';
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_SCENE_EDIT}_FAIL`,
        loading: false,
        payload: { error }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_SCENE_EDIT}_SUCCESS`, () => {
    expect(
      reducer(
        {
          item: {
            field_scenes: [
              {
                title: 'old title',
                field_slug: 'my-test'
              }
            ]
          }
        },
        {
          type: `${OPEN_EXPERIENCE_SCENE_EDIT}_SUCCESS`,
          loading: false,
          error: null,
          payload: {
            title: 'test',
            body: {
              value: 'test description'
            },
            field_slug: 'my-test'
          }
        }
      )
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_COMPONENT_FIELD_PRESAVE}_SUCCESS`, () => {
    expect(
      reducer(
        {
          item: {
            field_scenes: [
              {
                field_slug: 'testScene',
                field_components: [
                  {
                    id: 'testComponent',
                    title: 'My Old Value'
                  }
                ]
              }
            ]
          }
        },
        {
          type: `${OPEN_EXPERIENCE_COMPONENT_FIELD_PRESAVE}_SUCCESS`,
          loading: false,
          error: null,
          payload: {
            sceneSlug: 'testScene',
            component: 'testComponent',
            fieldName: 'title',
            fieldValue: 'My New Value'
          }
        }
      )
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_COMPONENT_EDIT}_LOADING`, () => {
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_COMPONENT_EDIT}_LOADING`,
        loading: true,
        error: null,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_COMPONENT_EDIT}_FAIL`, () => {
    const error = 'Error: failed to update component.';
    expect(
      reducer(undefined, {
        type: `${OPEN_EXPERIENCE_COMPONENT_EDIT}_FAIL`,
        loading: false,
        payload: { error }
      })
    ).toMatchSnapshot();
  });

  it(`Should handle ${OPEN_EXPERIENCE_COMPONENT_EDIT}_SUCCESS`, () => {
    const id = 10;
    const sceneSlug = 'testScene';
    expect(
      reducer(
        {
          item: {
            field_scenes: [
              {
                field_slug: sceneSlug,
                field_components: [
                  {
                    id,
                    title: 'My Old Value'
                  }
                ]
              }
            ]
          }
        },
        {
          type: `${OPEN_EXPERIENCE_COMPONENT_EDIT}_SUCCESS`,
          loading: false,
          error: null,
          payload: {
            id,
            sceneSlug,
            field_body: 'test body',
            title: 'My New Slug',
            field_x: 0,
            field_y: 0,
            field_z: 0
          }
        }
      )
    ).toMatchSnapshot();
  });
});
