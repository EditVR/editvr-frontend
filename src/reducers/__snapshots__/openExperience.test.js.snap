// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_COMPONENT_EDIT_FAIL 1`] = `
Object {
  "error": "Error: failed to update component.",
  "item": Object {},
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_COMPONENT_EDIT_LOADING 1`] = `
Object {
  "error": null,
  "item": Object {},
  "loading": true,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_COMPONENT_EDIT_SUCCESS 1`] = `
Object {
  "error": null,
  "item": Object {
    "field_scenes": Array [
      Object {
        "field_components": Array [
          Object {
            "field_body": "test body",
            "field_x": 0,
            "field_y": 0,
            "field_z": 0,
            "id": 10,
            "title": "My New Slug",
          },
        ],
        "field_slug": "testScene",
      },
    ],
  },
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_COMPONENT_FIELD_PRESAVE_SUCCESS 1`] = `
Object {
  "error": null,
  "item": Object {
    "field_scenes": Array [
      Object {
        "field_components": Array [
          Object {
            "id": "testComponent",
            "title": "My New Value",
          },
        ],
        "field_slug": "testScene",
      },
    ],
  },
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_FETCH_FOR_USER_FAIL 1`] = `
Object {
  "error": "Error: failed to fetch experiences.",
  "item": Object {},
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_FETCH_FOR_USER_LOADING 1`] = `
Object {
  "error": null,
  "item": Object {},
  "loading": true,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_FETCH_FOR_USER_SUCCESS 1`] = `
Object {
  "error": null,
  "item": Object {
    "id": 1,
    "title": "example experience",
  },
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_SCENE_CREATE_FAIL 1`] = `
Object {
  "error": "Error: failed to create scene.",
  "item": Object {},
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_SCENE_CREATE_LOADING 1`] = `
Object {
  "error": null,
  "item": Object {},
  "loading": true,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_SCENE_CREATE_SUCCESS 1`] = `
Object {
  "error": null,
  "item": Object {},
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_SCENE_EDIT_FAIL 1`] = `
Object {
  "error": "Error: failed to update scene.",
  "item": Object {},
  "loading": false,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_SCENE_EDIT_LOADING 1`] = `
Object {
  "error": null,
  "item": Object {},
  "loading": true,
}
`;

exports[`reducers->openExperience Should handle OPEN_EXPERIENCE_SCENE_EDIT_SUCCESS 1`] = `
Object {
  "error": null,
  "item": Object {
    "field_scenes": Array [
      Object {
        "body": "test description",
        "field_slug": "my-test",
        "title": "test",
      },
    ],
  },
  "loading": false,
}
`;

exports[`reducers->openExperience Should return the initial state by default 1`] = `
Object {
  "error": null,
  "item": Object {},
  "loading": false,
}
`;
